### PR TITLE
Use default version of OpenCV on current platform + Use catkin default c++ standard

### PR DESCRIPTION
**Pull request by Samahu**
## Linked Issues:
| EPIC ⚡ | [OCEANWATER-577 / Support for ROS Noetic](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-577) |
| :----------- | :----------- |
| Jira Ticket 🎟️   | [OCEANWATER-578 / Remove strict requirement on OpenCV 3.0 in ow_dynamic_terrain ](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-578) |
| Jira Ticket 🎟️   | [OCEANWATER-593 / Don't explicitly set a required C++ standard unless absolutely needed ](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-593) |

## Summary of Changes
This is a single iteration at our effort to support **ROS Noetic**, these following changes are necessary and does't break **ROS Melodic** compilation: 
* Use default version of OpenCV on current platform 
* Use catkin default c++ standard

## Test
* Make sure the system still successfully compiles against **ROS Melodic**
* If you happen to have an Ubuntu 20.04 system with all the prerequisites for OceanWATERS then you may verify that ow_dynamic_terrain doesn't fail due to a requiring OpenCV 3.0

### CODE REVIEW

File: ow_dynamic_terrain/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,6 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(ow_dynamic_terrain)
 
-## Compile as C++11, supported in ROS Kinetic and newer
-# add_compile_options(-std=c++11)
-add_compile_options(-std=c++11)
```
Comment (kmdalal) [inline ow_dynamic_terrain/CMakeLists.txt:6]: Just curious: is the removal of this add_compile_options and the others a requirement for Noetic, or an unrelated cleanup?

**Review by kmdalal — APPROVED** [review]
I'm unable to test this in Ubuntu 20 anytime soon, but the regression passed.

File: ow_dynamic_terrain/CMakeLists.txt (hunk 1 above)
Comment (Samahu) [reply ow_dynamic_terrain/CMakeLists.txt:6]: I am not sure why this was added in first place (probably the default template by catkin adds that) it might have been required at some point. Both **melodic** and **noetic** are currently embracing c++14 standard so having this module requesting c++11 resulted in a set of compile time errors for **noetic** (Ubuntu 20.04 has an upgraded version of the c/c++ compiler and libraries).

**Review by mogumbo — APPROVED** [review]
This compiles and runs on Ubuntu 18.04 with melodic. ow_dynamic_terrain doesn't compile on Ubuntu 20.04 with noetic but not because of a missing OpenCV 3.0.

**Comment by Samahu** [comment]
> This compiles and runs on Ubuntu 18.04 with melodic. ow_dynamic_terrain doesn't compile on Ubuntu 20.04 with noetic but not because of a missing OpenCV 3.0.

Yes, there are more problems down the line.
